File: g4/src/main/java/model/Teacher.java
package model;

import javax.persistence.Entity;
import java.util.ArrayList;

@Entity
public class Teacher extends User {
    ArrayList<Question> questions;
    ArrayList<Exam> exams;
    Integer points;
    Teacher(){
        points = 0;
    }
    //TODO FORMULA DE PUNTAJE
    public ArrayList<Question> getQuestions() {
        return questions;
    }

    public void setQuestions(ArrayList<Question> questions) {
        this.questions = questions;
    }

    public ArrayList<Exam> getExams() {
        return exams;
    }

    public void setExams(ArrayList<Exam> exams) {
        this.exams = exams;
    }
}

File: EasyExam/Front/EasyExam/src/app/shared/classes/teacher.ts
import { User } from '../interfaces/user';
import { Exam } from '../interfaces/exam';
import { Question } from './question';

export class Teacher implements User {
  email: string;
  lastName: string;
  name: string;
  password: string;
  phone: string;
  exams: Exam[];
  isAdmin: boolean;
  points: number;
  questions: Question[];
  reportedQuestions: Question[];
}
